### PR TITLE
fix: Correcting browser history and adding back button integration

### LINKED ISSUE #1998: Deeplinking (and address bar updating based on navigation) isnt working correctly right now. Let's move this issue across to extensions
Navigating using the extensions does not update the address bar with the current route

### PR DESCRIPTION
**Pull request by nickrandolph**
GitHub Issue (If applicable): fixes #1998, #2038

## PR Type

What kind of change does this PR introduce?
- Bugfix
- Feature

## What is the current behavior?

No support for back button to drive navigation
Updating address bar adds to browser history, which is meaningless if use navigates back within the app (which will continue to add items instead of removing them)

## What is the new behavior?

Address bar support is consistent with the SystemNavigationManager implementation - single item that is updated if there is somewhere to navigate back to
Added back button integration for browser
- back button is enabled if navigation can go back (ie in a frame within the nav hierarchy)
- back button is enabled if there's a previous site (eg you navigate from another site to the wasm application)

Both address bar updating and back button usage can be controlled using the NavigationConfiguration class that can be set in appsettings.json

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md). (for bug fixes / features) 
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Extensions.Navigation.UI/NavigatorExtensions.cs [hunk 1]
```diff
@@ -48,16 +48,16 @@ public static class NavigatorExtensions
...
+	private static async Task<INavigator?> GoBackNavigator(this INavigator navigator)
 	{
 		var dispatcher = navigator.Get<IServiceProvider>()!.GetRequiredService<IDispatcher>();
 
 		// Default to navigating back on the current navigator
 		if (await navigator.CanNavigate(new Route(Qualifiers.NavigateBack)))
-			//is ControlNavigator ctrlNav &&
-			//await dispatcher.ExecuteAsync(async ct => ctrlNav.CanGoBack))
+		//is ControlNavigator ctrlNav &&
+		//await dispatcher.ExecuteAsync(async ct => ctrlNav.CanGoBack))
```
Comment (jeromelaban) [inline src/Uno.Extensions.Navigation.UI/NavigatorExtensions.cs:58]: Still needed ?

File: src/Uno.Extensions.Navigation.UI/NavigatorExtensions.cs [hunk 2]
```diff
@@ -66,8 +66,25 @@ public static class NavigatorExtensions
...
-			new NavigationResponse(Success: false);
+		return gobackNavigator;
 	}
+
+	/// <summary>
+	/// Returns value whether any of the current INavigator instances supports navigating back
+	/// </summary>
+	/// <param name="navigator">The highest level navigator</param>
+	/// <returns>bool indicating whether go back is supported</returns>
+	public static async Task<bool> CanGoBack(this INavigator navigator)
```
Comment (jeromelaban) [inline src/Uno.Extensions.Navigation.UI/NavigatorExtensions.cs:77]: Is any new documentation needed?

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.Extensions.Navigation.UI/NavigatorExtensions.cs (hunk 1 above)
Comment (nickrandolph) [reply src/Uno.Extensions.Navigation.UI/NavigatorExtensions.cs:58]: No - will do a clean up PR separately for this and to add documentation to navigation apis

File: src/Uno.Extensions.Navigation.UI/NavigatorExtensions.cs (hunk 2 above)
Comment (nickrandolph) [reply src/Uno.Extensions.Navigation.UI/NavigatorExtensions.cs:77]: yes. There currently isn't proper documentation on the various navigation extension methods

**Comment by nickrandolph** [comment]
@mergify backport stable/release/4.0

**Comment by nickrandolph** [comment]
@mergify backport release/stable/4.0

**Pull request merged**